crates/nu-cli/src/completions/completer.rs

```diff
@@ -141,18 +141,24 @@ impl NuCompleter {
                             let current_span = working_set.get_span_contents(flat.0).to_vec();
                             let current_span_str = String::from_utf8_lossy(&current_span);
 
+                            let is_last_span = pos >= flat.0.start && pos < flat.0.end;
+
                             // Skip the last 'a' as span item
-                            if flat_idx == flattened.len() - 1 {
-                                let mut chars = current_span_str.chars();
-                                chars.next_back();
-                                let current_span_str = chars.as_str().to_owned();
-                                spans.push(current_span_str.to_string());
+                            if is_last_span {
+                                let offset = pos - flat.0.start;
+                                if offset == 0 {
+                                    spans.push(String::new())
+                                } else {
+                                    let mut current_span_str = current_span_str.to_string();
+                                    current_span_str.remove(offset);
+                                    spans.push(current_span_str);
```

Comment:
The offset is derived from pos and flat.0.start.
Neither of their values can possibly fall in the middle of a multibyte character, right?
So the offset is safe for strings that contain multibyte characters.

Reply:
Right they should be sound byte indexes on char boundaries by convention.

I was mentally hung up on the `.remove` thinking it was operating on just bytes like in the parser but yes here it would remove a multibyte char if pointed to with the correct byte index and panic otherwise.

So a bug to trigger this panic would have to be somewhere else and we don't have to be too defensive.